src/sql/_testing.py

```diff
@@ -25,6 +26,21 @@
 TMP_DIR = "tmp"
 
 
+class TestingShell(InteractiveShell):
+    """
+    A custom InteractiveShell that raises exceptions instead of silently suppressing
+    them.
+    """
+
+    def run_cell(self, *args, **kwargs):
```

Comment:
Is it not needed for the other functions like `run_line_magic`, `run_cell_magic` etc?

Reply:
good catch. I tried to patch the others but encounter some issues: they rely on one another and behave differently. so it isn't as simple as patching them all. I can tackle this in the next PR. most of the tests use `run_cell`, only a few of them use other methods https://github.com/ploomber/jupysql/issues/760